### DISCUSSION TITLE
Clear screen with a keybinding

### DISCUSSION THREAD

**Discussion by awmaMe** [ohmyzsh/ohmyzsh#11557] [Help] [answered]
I can clear the screen with Ctrl+L, but I would like to clear the screen entirely just like running the clear command.
I have tried adding this line in .zshrc:
`bindkey "^L" clear-screen`

I am still getting the same behavior (clears the screen, but can scroll up to previous outputs)

**Comment by mcornella** [comment, marked as answer]
You can define your own zle widget that calls `tput reset`, which clears everything in the terminal:

```zsh
clear-terminal() { tput reset; zle redisplay; }
zle -N clear-terminal
bindkey '^L' clear-terminal
```

**Reply by awmaMe** [reply]
Thank you!

**Reply by tutkun** [reply]
Hi @mcornella ,

How can I use the `tput reset;` effect for the `cmd+k` command on macOS?
Currently, when I press `cmd+k` in `iterm2`, it distorts the two-line terminal view by lowering it to a single line.

It should be :
<img width="255" height="108" alt="Image 1" src="https://github.com/user-attachments/assets/3a099af9-f811-46d0-9ce4-0b1bd754dfad" />

But, it is :
<img width="262" height="98" alt="Image 2" src="https://github.com/user-attachments/assets/1cec19df-534d-4bd8-a3bf-0d46a9d9bcb9" />
